athena-vertica/src/main/java/com/amazonaws/athena/connectors/vertica/VerticaRecordHandler.java

```diff
@@ -330,28 +334,24 @@ public HashMap<String, Object> getNameValue() {
         }
     }
 
-
-    /*
-    Method to create the Parquet read request
-     */
-    private static SelectObjectContentRequest generateBaseParquetRequest(String bucket, String key)
+    @VisibleForTesting
+    protected ArrowReader constructArrowReader(String uri)
     {
-        SelectObjectContentRequest request = new SelectObjectContentRequest();
-        request.setBucketName(bucket);
-        request.setKey(key);
-        request.setExpression(VerticaRecordHandler.QUERY);
-        request.setExpressionType(ExpressionType.SQL);
-
-        InputSerialization inputSerialization = new InputSerialization();
-        inputSerialization.setParquet(new ParquetInput());
-        inputSerialization.setCompressionType(CompressionType.NONE);
-        request.setInputSerialization(inputSerialization);
-
-        OutputSerialization outputSerialization = new OutputSerialization();
-        outputSerialization.setJson(new JSONOutput());
-        request.setOutputSerialization(outputSerialization);
+        BufferAllocator allocator = new RootAllocator();
+        DatasetFactory datasetFactory = new FileSystemDatasetFactory(
+                allocator,
+                NativeMemoryPool.getDefault(),
+                FileFormat.PARQUET,
+                uri);
+        Dataset dataset = datasetFactory.finish();
+        ScanOptions options = new ScanOptions(/*batchSize*/ 32768);
```

Comment:
how do we determine scan batch size?

Reply:
This is just the value used in all of the Apache Arrow dataset documentation. I am open to suggestions for other values
https://arrow.apache.org/docs/dev/java/dataset.html